floss/language_identifier.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+import pefile
+
+import floss.logging_
+from floss.logging_ import DebugLevel
+
+logger = floss.logging_.getLogger("floss")
+
+
+def is_go_bin(sample: str) -> bool:
+    """
+    Check if the binary given is compiled with GO compiler or not
+    it checks the magic header of the pclntab structure -pcHeader-
+    the magic values varies through the version
+    reference:
+    https://github.com/0xjiayu/go_parser/blob/865359c297257e00165beb1683ef6a679edc2c7f/pclntbl.py#L46
+    """
+    try:
+        pe = pefile.PE(sample)
+    except:
+        logger.debug("NOT valid PE header")
+        return False
+
+    go_magic = [
+        b"\xf0\xff\xff\xff\x00\x00",
+        b"\xfb\xff\xff\xff\x00\x00",
+        b"\xfa\xff\xff\xff\x00\x00",
+        b"\xf1\xff\xff\xff\x00\x00",
+    ]
+
+    # look for the .rdata section first
+    for section in pe.sections:
+        if ".rdata" in section.Name.decode("utf-8").rstrip("\x00"):
```

Comment:
please wrap this in a try/except because this decoding might easily fail on corrupt PE files
```suggestion
        try:
            section_name = section.Name.partition(b"\x00")[0].decode("utf-8")
        except UnicodeDecodeError:
            continue
        if ".rdata" in section_name:
```